system/ee/legacy/helpers/file_helper.php

```diff
@@ -72,23 +72,27 @@ function write_file($path, $data, $mode = FOPEN_WRITE_CREATE_DESTRUCTIVE)
  * If the second parameter is set to TRUE, any directories contained
  * within the supplied base directory will be nuked as well.
  *
- * @access	public
- * @param	string	$path		Path to file
- * @param	bool	$del_dir	Whether to delete any directories found in the path
- * @param	int		$level		Levels deep to traverse the file tree to delete files
- * @param	array	$exclude	Array of file names to exclude from deletion
- * @return	bool
+ * @access  public
+ * @param   string  $path       Path to file
+ * @param   bool    $del_dir    Whether to delete any directories found in the path
+ * @param   int     $level      Levels deep to traverse the file tree to delete files
+ * @param   array   $exclude    Array of file names to exclude from deletion
+ * @return  bool
  */
 if (! function_exists('delete_files')) {
     function delete_files($path, $del_dir = false, $level = 0, $exclude = array())
     {
         // Trim the trailing slash
         $path = rtrim($path, DIRECTORY_SEPARATOR);
 
-        if (! $current_dir = @opendir($path)) {
+        // If this isnt a directory, lets return false
+        if (! is_dir($path)) {
             return false;
         }
 
+        // Get the current directory
+        $current_dir = @opendir($path);
```

Comment:
I think this might cause issues when the directory exists, but the permissions are incorrect. I'll push an update